src/lib/djerba/plugins/tar/provenance_tools.py

```diff
@@ -0,0 +1,77 @@
+"""
+Supporting functions to pull files from provenance
+"""
+import csv
+from decimal import Decimal
+import logging
+import re
+
+import djerba.util.provenance_index as index
+import djerba.plugins.tar.sample.constants as constants
+
+
+def subset_provenance(self, workflow, root_sample_name):
+    provenance_location = 'provenance_subset.tsv.gz'
+    provenance = []
+    try:
+        with self.workspace.open_gzip_file(provenance_location) as in_file:
+            reader = csv.reader(in_file, delimiter="\t")
+            for row in reader:
+                if row[index.WORKFLOW_NAME] == workflow and row[index.ROOT_SAMPLE_NAME] == root_sample_name:
+                    provenance.append(row)
+    except OSError as err:
+        msg = "Provenance subset file '{0}' not found when looking for {1}".format('provenance_subset.tsv.gz', workflow)
+        raise RuntimeError(msg) from err
+    return(provenance)
+
+def subset_provenance_sample(self, workflow, input_group_id, suffix):
```

Comment:
do you need both `subset_provenance_sample` and `subset_provenance` ? If yes, might be good to add a comment about the difference - if no, remove